### PR TITLE
[CALCITE-6725] The caching mechanism key in ElasticsearchSchemaFactory is affected by the order of hosts

### PR DESCRIPTION
**Pull request by AstraHoo**

### CODE REVIEW

**Comment by mihaibudiu** [comment]
Please add at least one unit test, ideally showing that a bug no longer exists.

File: elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/ElasticSearchAdapterTest.java [hunk 1]
```diff
@@ -771,4 +771,24 @@ private static Consumer<ResultSet> sortedResultSetChecker(String column,
             "state=AR; EXPR$1=3; EXPR$2=3");
   }
 
+    @Test
+    void testSortHosts() {
```
Comment (caicancai) [inline elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/ElasticSearchAdapterTest.java:775]: Add jira link

File: elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/ElasticSearchAdapterTest.java [hunk 2]
```diff
@@ -771,4 +771,24 @@ private static Consumer<ResultSet> sortedResultSetChecker(String column,
...
+      Map<String, Integer> hosts = new HashMap<>();
+      hosts.put("192.168.1.150",8080);
+      hosts.put("192.168.1.110",6080);
+      hosts.put("192.168.1.120",8080);
+
+
+      List<HttpHost> sortedHosts = hosts.entrySet()
```
[the remark this answers is not in the document]
Comment (mihaibudiu) [reply elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/ElasticSearchAdapterTest.java:782]: What @NobiGo means here is: you are testing the sorting method, but not the elastic search. Ideally a unit test would show a bug related to elastic search that is solved by this change.

**Review by mihaibudiu — APPROVED** [review]
please squash the commits - use the same commit message as the jira title

**Review by caicancai — APPROVED** [review]

**Pull request closed without merging**